index.bs

```diff
@@ -795,6 +743,562 @@ the following list:
+    1. Set [=this=]'s [=WebSocketStream/handshake aborted flag=].
+ 1. Let |client| be [=this=]'s [=relevant settings object=].
+ 1. Run this step [=in parallel=]:
+  1. [=Establish a WebSocket connection=] given |urlRecord|, |protocols|, and |client|. [[!FETCH]]
+
+     <p class="note">If the [=establish a WebSocket connection=] algorithm fails, it triggers the
+     [=fail the WebSocket connection=] algorithm, which then invokes the [=close the WebSocket
+     connection=] algorithm. This establishes that [=the WebSocket connection is closed=], which
+     <a>rejects</a> the {{WebSocketStream/opened}} and {{WebSocketStream/closed}} promises.
+</div>
+
+<hr>
+
+The <dfn attribute for=WebSocketStream>url</dfn> getter steps are to return [=this=]'s
+[=WebSocketStream/url=], [=URL serializer|serialized=].
+
+The <dfn attribute for=WebSocketStream>opened</dfn> getter steps are to return [=this=]'s
+[=WebSocketStream/opened promise=].
+
+The <dfn attribute for=WebSocketStream>closed</dfn> getter steps are to return [=this=]'s
+[=WebSocketStream/closed promise=].
+
+<div algorithm>
+ The <dfn method for=WebSocketStream>close(|closeInfo|)</dfn> method steps are:
+
+ 1. Let |code| be |closeInfo|["{{WebSocketCloseInfo/closeCode}}"] if present, or null otherwise.
+ 1. Let |reason| be |closeInfo|["{{WebSocketCloseInfo/reason}}"].
+ 1. [=Close the WebSocket=] with <var ignore>this</var>, |code|, and |reason|.
+</div>
+
+
+## Feedback to WebSocketStream from the protocol ## {#feedback-to-websocket-stream-from-the-protocol}
+
+When [=the WebSocket connection is established=] for a {{WebSocketStream}} |stream|, the user agent
+must [=queue a task=] to run these steps:
+
+<div algorithm="handle WebSocketStream connection establishment">
+
+ 1. Change the [=WebSocketStream/ready state=] to {{WebSocket/OPEN}} (1).
+ 1. Set |stream|'s [=WebSocketStream/was ever connected flag=].
+ 1. Let |extensions| be the [=extensions in use=].
+ 1. Let |protocol| be the [=subprotocol in use=].
+ 1. Let |pullAlgorithm| be an action that [=pulls bytes=] from |stream|.
+ 1. Let |cancelAlgorithm| be an action that [=cancels=] the {{WebSocketStream}} with |stream| and
+    |reason|, given |reason|.
+ 1. Let |readable| be a [=new=] {{ReadableStream}}.
+ 1. [=ReadableStream/Set up=] |readable| with |pullAlgorithm| and |cancelAlgorithm|.
+ 1. Let |writeAlgorithm| be an action that [=writes=] |chunk| to |stream|, given |chunk|.
+ 1. Let |closeAlgorithm| be an action that [=closes=] |stream|.
+ 1. Let |abortAlgorithm| be an action that [=aborts=] |stream| with |reason|, given |reason|.
+ 1. Let |writable| be a [=new=] {{WritableStream}}.
+ 1. [=WritableStream/Set up=] |writable| with |writeAlgorithm|, |closeAlgorithm|, and
+    |abortAlgorithm|.
+ 1. Set |stream|'s [=WebSocketStream/readable stream=] to |readable|.
+ 1. Set |stream|'s [=WebSocketStream/writable stream=] to |writable|.
+ 1. Let |openInfo| be a [=new=] {{WebSocketOpenInfo}}.
```

Comment:
You don't use [=new=] on dictionaries. Probably it's nicest to combine this into something like

> Let _openInfo_ be the `WebSocketOpenInfo` «[ ...

or maybe even

> Resolve _stream_'s opened promise with the `WebSocketOpenInfo` «[ ...

Reply:
Thanks, done.